db/src/tx_observer.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use std::sync::{
+    Arc,
+};
+use std::thread;
+
+use indexmap::{
+    IndexMap,
+};
+
+use types::{
+    AttributeSet,
+    TxReport,
+};
+
+pub struct TxObserver {
+    notify_fn: Arc<Option<Box<Fn(String, Vec<TxReport>) + Send + Sync>>>,
```

Comment:
I'm not sure why there's an `Option` in here. Can you explain?

Reply:
I think that was a leftover from a previous experiment.